.github/workflows/scala_formatting.yml

```diff
@@ -0,0 +1,51 @@
+# Runs auto-formatting script on push to any branch
+name: "Run auto formatting"
+
+# Unsure whether these need to be on the caller or called workflow
+permissions:
+  id-token: write
+  contents: write
+
+on: workflow_call
+
+jobs:
+  autoformat:
+    name: autoformat
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out project
+        uses: actions/checkout@v4
+
+      - name: Configure AWS credentials
+        id: aws_credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: eu-west-1
+          role-to-assume: ${{ secrets.GHA_SCALA_FORMATTING_ROLE_ARN }}
+          output-credentials: true
+
+      - name: Log in to private ECR
+        uses: aws-actions/amazon-ecr-login@v2
+
+      - name: Run formatting
+        run: bash ./builds/run_formatting.sh
+        env:
+          AWS_ACCESS_KEY_ID: ${{ steps.aws_credentials.outputs.aws-access-key-id }}
+          AWS_SECRET_KEY: ${{ steps.aws_credentials.outputs.aws-secret-access-key }}
+          AWS_SESSION_TOKEN: ${{ steps.aws_credentials.outputs.aws-session-token }}
+
+      - name: Check for formatting changes
+        id: check_formatting_changes
+        run: |
+          if [[ -n $(git status --porcelain) ]]; then
+              echo "changes=true" >> "$GITHUB_OUTPUT"; 
+          fi
+
+      - name: Commit and push formatting changes
+        if: steps.check_formatting_changes.outputs.changes == 'true'
+        run: |
+          git config user.name "Github on behalf of Wellcome Collection"
+          git config user.email "wellcomedigitalplatform@wellcome.ac.uk"
+          git commit -am "Apply auto-formatting rules"
```

Comment:
Will that also prevent the build triggered by the commit before "Apply auto-formatting rules" from being cancelled?
Right now it's:
- commit push triggers bk build and formatting workflow run
- if there are formatting changes, formatting workflow pushes a new commit. This cancels the above bk build
- formatting changes triggers a new bk build
If we don't do the last step, we're left with a cancelled build?

Reply:
Nice, the previous build keeps running after the formatting commit is pushed 👍
The only things is, the tests could fail in the `HEAD~1` commit but it looks on the PR like all checks have passed because it only shows the checks run on the last commit.